tests/ssl-opt.sh

```diff
@@ -5360,7 +5360,6 @@ run_test    "Authentication: server goodcert, client required, no trusted CA" \
 # occasion (to be fixed). If that bug's fixed, the test needs to be altered to use a
 # different means to have the server ignoring the client's supported curve list.
 
```

Comment:
Shouldn't we require ECP_LIGHT here and below instead?

Reply:
Good question. I don't think we should require `ECP_LIGHT` here: we already auto-detect based on using server5 that ECDSA support is required. Now perhaps we need to reflect somewhere that PK only has support for ECDSA when `ECP_LIGHT` is present. Let me check.

Reply:
Ok, I checked and I think there are multiple sides to this:

1. Definition of `MBEDTLS_PK_CAN_ECDSA_xxx` macros in `pk.h`.
2. Definition of `MBEDTLS_PK_HAVE_ECDSA` helper macro in `check_config.h`.
3. Definition of `requires_pk_alg()` in `ssl-opt.sh`.

None of them are fully correct right now - that is, neither in this PR nor in development. Before this PR, in the `USE_PSA` case, we should require `ECP_C` in addition to `PSA_WANT_ALG_ECDSA` because without it, PK doesn't support ECC keys; after this PR it should be `ECP_LIGHT`.

Though since we don't want to make `ECP_LIGHT` public, again what we should do it not require it, but instead auto-enable it when needed, so for example `PK_C && USE_PSA && PSA_WANT_ALG_ECDSA` would auto-enable `ECP_LIGHT` and then I think all three of the above definitions would become correct.

Reply:
See #7442.